### PR TITLE
Reduce disk space taken by JUnit test reports on CI

### PR DESCRIPTION
**Pull request by yrodiere**
This attempts to take advantage of https://github.com/jenkinsci/junit-plugin/pull/601

I'm not sure `stdioRetention` is available in the pipeline syntax though, so... let's see how this PR goes on CI.

I wish there was a way to set defaults for this globally :/

### CODE REVIEW

**Review by marko-bekhta — APPROVED** [review]
🤞🏻

File: Jenkinsfile [hunk 1]
```diff
@@ -878,6 +878,12 @@ void withMavenWorkspace(Closure body) {
 }
 
 void withMavenWorkspace(Map args, Closure body) {
+	args.putIfAbsent("options", [
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply Jenkinsfile]: It shouldn't happen, but you're right. I addressed this.

[the remark this answers is not in the document]
Comment (yrodiere) [reply Jenkinsfile]: Jenkins's CPS stuff is giving me a hard time. I'll just use `put`.

**Comment by yrodiere** [comment]
Test failures seem unrelated... ?

Anyway... this didn't work:

```
WARNING: Unknown parameter(s) found for class type 'org.jenkinsci.plugins.pipeline.maven.publishers.JunitTestsPublisher': stdioRetention
```

I'll need to explore other options :/

**Comment by yrodiere** [comment]
FWIW the pipeline-maven-plugin hasn't even been upgraded to the version of `jenkins-junit` that exposes ``: https://github.com/jenkinsci/pipeline-maven-plugin/pull/762

So we're a long way of being able to configure this through pipelines.

That being said... Guillaume is currently setting up something to push build scans of Hibernate Search to https://ge.hibernate.org/scans . Perhaps this would suit all our needs, and we would be able to disable the `JunitTestsPublisher` completely?

I will have a look at some point.

**Comment by yrodiere** [comment]
Superseded by #3994

**Pull request closed without merging**